### PR TITLE
[Papercut][SW-13559] Fix tax display

### PR DESCRIPTION
**Pull request by htkassner**
## Description

Please describe your pull request:
- Why is it necessary? Fixes display of tax for numbers without decimals and for german locale
- What does it improve? look & feel
- Does it have side effects? No

| Questions | Answers |
| --- | --- |
| BC breaks? | no |
| Tests pass? | yes |
| Related tickets? | https://issues.shopware.com/#/issues/SW-13559 |
| How to test? | Just checkout a product and watch taxes display. Until now in german language a floating point number with a point character was shown. Now it shows no decimals for non decimals and a comma in german tax display. |

### CODE REVIEW

File: themes/Frontend/Bare/frontend/checkout/cart_footer.tpl [hunk 1]
```diff
@@ -139,7 +139,13 @@
                 {block name='frontend_checkout_cart_footer_field_labels_taxes'}
                     {if $sUserData.additional.charge_vat}
                         {foreach $sBasket.sTaxRates as $rate => $value}
-
+                            {if ($rate * 100) % 100 == 0}
```
Comment (bcremer) [inline themes/Frontend/Bare/frontend/checkout/cart_footer.tpl]: `($rate * 100) % 100 == 0` is this supposed to be a `isFloat` kind of check?

File: themes/Frontend/Bare/frontend/checkout/cart_footer.tpl [hunk 2]
```diff
@@ -139,7 +139,13 @@
...
+                                {$rate = $rate|number_format}
+                            {else}
+                                {if $Locale == "de_DE"}
+                                    {$rate = $rate|number_format:2:",":"."}
```
Comment (bcremer) [inline themes/Frontend/Bare/frontend/checkout/cart_footer.tpl]: Should't this be solved generally with a locale aware number formatter?
There is https://github.com/shopware/shopware/blob/5.2/engine/Library/Enlight/Template/Plugins/modifier.number.php#L33 but I am not sure if that solves your problem.

**Comment by htkassner** [comment]
Gonna recap this. Didn´t know there was a custom modifier!

**Comment by htkassner** [comment]
Last commit is more general, uses existing functions and also unifies tax presentation for pdf documents.

File: snippets/documents/index.ini [hunk 3]
```diff
@@ -18,7 +18,7 @@ DocumentIndexOrderID = "Order No.:"
 DocumentIndexPageCounter = "Page {$page+1} of {$Pages|@count}"
 DocumentIndexSelectedDispatch = "Selected shipping type"
 DocumentIndexSelectedPayment = "Selected payment method"
-DocumentIndexTax = "Plus {$key}% VAT:"
+DocumentIndexTax = "Plus {if !$key|is_float}{$key = $key|number_format}{else}{$key = $key|number}{/if}{$key} % VAT:"
```
Comment (janbuecker) [inline snippets/documents/index.ini]: why don't you use this snippet? or isn't it working?

```
{if !$key|is_float}{$key|number_format}{else}{$key|number}{/if}
```

Comment (htkassner) [reply snippets/documents/index.ini]: Stupidness.

**Comment by bcremer** [comment]
I have the feeling the following line should be refactored into something re-usable:
`if (!$key|is_float}{$key|number_format}{else}{$key|number}{/if}`

What about extracting this logic into a separate smarty modifier so it's just `{$key|readable_number}` or `{$key|number:strip_superfluous_fraction}`. Please also add an Unit test for that.

The if float then check ist re-occurring logic that should not be duplicated in my opinion.

**Comment by htkassner** [comment]
Then we should do something like we did with currency and call it tax because this is espacially for taxes. So maybe implementing a helper and use it like $tax|tax.

**Comment by bcremer** [comment]
Please write an unit test for `smarty_modifier_tax`

File: snippets/documents/index.ini [hunk 4]
```diff
@@ -18,7 +18,7 @@ DocumentIndexOrderID = "Order No.:"
 DocumentIndexPageCounter = "Page {$page+1} of {$Pages|@count}"
 DocumentIndexSelectedDispatch = "Selected shipping type"
 DocumentIndexSelectedPayment = "Selected payment method"
-DocumentIndexTax = "Plus {$key}% VAT:"
+DocumentIndexTax = "Plus {$key|tax} % VAT:"
```
Comment (bcremer) [inline snippets/documents/index.ini]: We should add an non breaking space between the tax and the percent sign `&nbsp;`

**Pull request merged**